assets/src/application/ui/display/confirm/Confirm.tsx

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { __ } from '@wordpress/i18n';
+
+import { AlertDialog, AlertDialogProps } from '@infraUI/display';
+import ConfirmDelete from './ConfirmDelete';
+
+interface ConfirmProps extends AlertDialogProps {
+	type: 'delete';
+}
+
+const Confirm: React.FC<ConfirmProps> = ({ type, ...props }) => {
+	if (type === 'delete') {
+		return <ConfirmDelete {...props} />;
+	}
+
+	return <AlertDialog {...props} />;
+};
+
+export default Confirm;
```

Comment:
I would do this the other way around

Having a single "master" component like this will require a more complex API in order to configure things, which in turn makes client code more complex. It's better to have an abstracted "parent" component that hides the complexities away, and then create simple straightforward facades that take care of some additional configuration, so that client code can be as simple and clean as possible. (there are of course exceptions where having a single master component works better, like when you are dynamically generating content based on some other configuration)

So instead of importing a `Confirm` component into client code and then having to pass a prop to tell it you actually want a `ConfirmDelete` popup, it's easier, simpler, and more clear for the client code to just import the `ConfirmDelete` component directly. THEN, the `ConfirmDelete` component can internally use the `Confirm` component which internally uses the `AlertDialog` adapter.

Another reason to do things this way: currently the state management for controlling whether the `AlertDialog` is open or not is inside the adapter. But that logic does not belong there. That logic belongs in the app layer.

Here's what I want you to do:

- rename `Confirm` to `useConfirmDialog` (or maybe `useAlertDialog`?) - yes we are going to transform it into a hook

- move all of the state management logic and extra implementation details from `AlertDialog` into this new hook

- also move the actual import and configuration of the `AlertDialog` adapter into this new hook

- import the `useConfirm` hook into `ConfirmDelete` and `ConfirmClose` and whatever other confirm dialog components we need/want

- have these other components pass their configuration/implementation details to `useConfirmDialog` which will then return the configured and ready to go `AlertDialog`

- return the "configured and ready to go" `AlertDialog` for use in client code

This approach accomplishes the following:

- `AlertDialog` stays as a simple straightforward adapter that does nothing except decouple our application from third party code. If we need to change UI libraries in the future (or if there are breaking changes in the implementation of that same third party component) we are insulated from those changes, and only need to refactor the adapter to handle the external changes

- `useConfirmDialog` hides the complexities of state management and adapter configuration and can also handle things like swapping what components to use (like button vs link triggers) so that that logic does not have to be dealt with inside our other components.

- the above allows "child" components like `ConfirmDelete` and `ConfirmClose` to be much simpler because the hook can do most of the heavy lifting. For example, both of those "child" components could have a prop like `useLinkTrigger` and if set to true would pass a `LinkProps` object instead of ` `ButtonProps` object to `useConfirmDialog` which would use the presence of that object in a conditional to decide what trigger component to use. The "child" components could still have logic for exposing some of their configuration to client code for overriding a set of internal defaults, just like you have done in `ConfirmDelete` with the `title`

- client code requires far less configuration because 90% of the time, the "child" components should work just fine as is, but can be reconfigured via props for the edge cases. As well, since client code is dealing with the simplified and specialized "child" components, the APIs will be much easier to understand and manage as opposed to having to figure out how to get a single "master" Confirm component to spit out the correct thing you need.

- ideally, if we need/want to add a new type of confirm dialog component, we don't need to touch any other code. We simply need to import the `useConfirmDialog` hook into the new component and configure things the way we want. The way you are doing things now would require the `Confirm` component to be modified every time you want to add a new type of confirm dialog component (assuming that `Confirm` would allow this new component to be utilized via its `type` prop`).

so the final result should work something like this:

- the `AlertDialog` adapter is dead simple and is only used for managing differences in props between our app and the Chakra UI library. It does NOTHING else.

- `useConfirmDialog` handles all of the state management and as much shared logic as possible in order to hide that complexity and keep the final user APIs as simple as possible

- "child" components like `ConfirmDelete` and `ConfirmClose` utilize and configure `useConfirmDialog` via props, and provide a set of sensible defaults that can be exposed as needed to client code for further customization.

- client code utilizing the "child" components can hopefully in most cases not have to do anything in terms of customization because the sensible defaults in the child components work fine as is

Reply:
updated: https://github.com/eventespresso/event-espresso-core/pull/2682/commits/73646067896bc6a7278ffb59a47ef7d37c5db594
***
`ConfirmClose` will be added in this PR along with related logic
https://github.com/eventespresso/event-espresso-core/pull/2636